examples/xml-record-conversion/xml_record_conversion.bal

```diff
@@ -0,0 +1,58 @@
+import ballerina/io;
+import ballerina/xmldata;
+
+// Defines a record type with annotations.
+@xmldata:Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+type Invoice record {
+    string address;
+    Item[] items;
+    @xmldata:Attribute
+    string 'xmlns;
+    @xmldata:Attribute
+    string attr?;
+    @xmldata:Attribute
+    string ns\:attr?;
+};
+
+type Item record {
+    string|ItemCode itemCode;
+    int count;
+};
+
+@xmldata:Namespace {
+    uri: "http://example.com"
+}
+type ItemCode record {
+    @xmldata:Attribute
+    string discount;
+    string \#content;
+};
+
+// Creates an `Invoice` records.
+Invoice data = {
+    items: [
+        {itemCode: "223345", count: 10},
+        {itemCode: "223300", count: 7},
+        {
+            itemCode: {discount: "22%", \#content: "200777"},
+            count: 7
+        }
+    ],
+    address: "20, Palm grove, Colombo 3",
+    attr: "attr-val",
+    'xmlns: "http://sample.com",
+    ns\:attr: "http://attr.com"
+};
+
+public function main() returns error? {
+    // Converts a `record` representation to its XML representation.
+    xml result = check xmldata:toXml(data);
+    io:println(result);
+
+    // Converts an XML representation to its `record` representation.
+    Invoice output = check xmldata:fromXml(result);
+    io:println(output);
```

Comment:
This should be in two different BBEs. one for record -> XML, other for XML -> record

Reply:
We are maintaining a BBE for the conversion between  XML and JSON. So I also added one for this. 
https://github.com/ballerina-platform/ballerina-distribution/tree/master/examples/xml-json-conversion

Reply:
We need to rethink them later. We need to make them as simple as possible